pkg/server/api/bundle/v1/service.go

```diff
@@ -280,7 +280,9 @@ func (s *Service) BatchCreateFederatedBundle(ctx context.Context, req *bundlev1.
 		r := s.createFederatedBundle(ctx, b, req.OutputMask)
 		results = append(results, r)
 
-		rpccontext.AuditRPCWithTypesStatus(ctx, r.Status, fieldsFromBundleProto(b, nil))
+		if _, ok := rpccontext.AuditLog(ctx); ok {
```

Comment:
It is to avoid parsing to fields all entries requests when audit log is not enabled, 
It is not a big issue for regular endpoints but batch endpoints can contains several entries and parsing it when it is not required will consume unnecessary resources

Reply:
Ah I see. In that case, perhaps a cleaner way to handle this could be to have the audit log helpers in `rpccontext` accept a `func() logrus.Fields` rather than a `logrus.Fields`, and then have each RPC handler define a lambda for extracting the fields. That way we don't need extra conditionals in the API handler code.

Here is a simplified example of what I mean:

```go
package entry

type Entry struct {
    // Contains some fields
}

type CreateRequest struct {
    Entry *Entry
}

type UpdateRequest struct {
    Entry *Entry
}

func (h *Handler) CreateEntry(ctx context.Context, req *CreateEntryRequest) (*CreateEntryResponse, error) {
    auditFields := func() logrus.Fields {
        return fieldsFromEntry(req.Entry)
    }

    status, err := createEntry(ctx, req)
    if err != nil {
        return nil, err
    }

    rpccontext.AuditRPCWithTypesStatus(ctx, status, auditFields)

    ...
}

func (h *Handler) Update(ctx context.Context, req *UpdateEntryRequest) (*UpdateEntryResponse, error) {
    auditFields := func() logrus.Fields {
        return fieldsFromEntry(req.Entry)
    }

    status, err := updateEntry(ctx, req)
    if err != nil {
        return nil, err
    }

    rpccontext.AuditRPCWithTypesStatus(ctx, status, auditFields)

    ...
}

func fieldsFromEntry(entry Entry) logrus.Fields {
    f := logrus.Fields{}
    // Populate fields from entry
    ...
    return f
}
```

In `rpccontext`:
```go
package rpccontext

func AuditRPCWithTypesStatus(ctx context.Context, s types.Status, fieldsFunc func() logrus.Fields) {
    if auditLog, ok := AuditLog(ctx); ok {
        auditLog.AuditWithTypesStatus(fieldsFunc(), s)
    }
}
```

Reply:
agree I was thinking in do that I'll apply it now

Reply:
Looks much cleaner now, thanks!